### PR TITLE
Update strategy flags, fix naming

### PR DESCRIPTION
**Pull request by nickbp**
- plan-strategy => phase_strategy
- placement-strategy => placement_strategy
- Fix a flag description

The "plan" -> "phase" rename invokes a new framework build

### CODE REVIEW

**Comment by kensipe** [comment]
tested  LGTM

**Comment by nickbp** [comment]
Rebases against the recent changes and adds some fixes. Looks like it's working great now (incl the name handling cleanup).

File: repo/packages/K/kafka/6/config.json [hunk 1]
```diff
@@ -51,433 +50,433 @@
...
           "kafka_advertise_host_ip":{
-            "description":"Override advertised.host.name with the IP address detected in /opt/mesosphere/detect_ip",
+            "description":"Automatic configuration of each Broker node's advertised.host.name with the IP address detected in /opt/mesosphere/detect_ip",
             "type":"boolean",
             "default":true
           },
-          "kafka_override_auto_create_topics_enable":{
-            "description":"Override auto.create.topics.enable: Enable auto creation of topic on the server",
+          "auto_create_topics_enable":{
+            "description":"auto.create.topics.enable: Enable auto creation of topic on the server",
```
Comment (kensipe) [inline repo/packages/K/kafka/6/config.json]: the property is auto_create_topics_enable and the description is for auto.create.topics.enable should these be aligned?

File: repo/packages/K/kafka/6/package.json [hunk 2]
```diff
@@ -7,6 +7,6 @@
   "framework": true,
   "tags": ["message", "broker", "pubsub"],
   "postInstallNotes": "DCOS Kafka Service is being installed.",
-  "postUninstallNotes": "DCOS Kafka Service has been uninstalled. Please run  the framework_cleaner to remove any persistent state if required."
+  "postUninstallNotes": "DCOS Kafka Service has been uninstalled. Please run the framework_cleaner to remove any persistent state if required."
```
Comment (kensipe) [inline repo/packages/K/kafka/6/package.json]: nice attention to detail

File: repo/packages/K/kafka/6/config.json (hunk 1 above)
Comment (nickbp) [inline repo/packages/K/kafka/6/config.json]: These settings are being passed to Kafka: DCOS/Marathon wants underscores, while Kafka wants periods

https://kafka.apache.org/08/configuration.html

File: repo/packages/K/kafka/6/package.json (hunk 2 above)
Comment (nickbp) [inline repo/packages/K/kafka/6/package.json]: Saw this during testing :)

File: repo/packages/K/kafka/6/config.json (hunk 1 above)
Comment (kensipe) [inline repo/packages/K/kafka/6/config.json]: makes sense!

Comment (kensipe) [inline repo/packages/K/kafka/6/config.json]: except... I believe that the config.json supports titles (at least it did).  so we could have a property auto_create_topics_enable with a "title" property of "auto.create.topics.enable" which would underscores for dcos marathon but dot notation for users and would align with kafka user expectations.

**Comment by kensipe** [comment]
I would love to see dot notation for the property names which I think can be implemented with the title property.  outside of that... I tested it and LGTM

**Pull request merged**